src/dbnode/network/server/tchannelthrift/node/service.go

```diff
@@ -246,6 +265,10 @@ func (s *service) Health(ctx thrift.Context) (*rpc.NodeHealthResult_, error) {
 // endpoint because while the Health endpoint provides the same information, this endpoint does not
 // require parsing the response to determine if the node is bootstrapped or not.
 func (s *service) Bootstrapped(ctx thrift.Context) (*rpc.NodeBootstrappedResult_, error) {
+	if s.db == nil {
```

Comment:
This is racey to read as well without an RLock, because you can't guarantee it's not currently being changed from nil to not nil as you read this (since no RLock held).